### PR TITLE
Remove Redundant Setting of OP_WRITE Interest

### PR DESCRIPTION
**Pull request by original-brownbear**
* We shouldn't have to set OP_WRITE interest before running into a partial write. Since setting OP_WRITE is handled by the `eventHandler.postHandling` logic, I think we can simply remove this operation and simplify/remove tests that were testing the setting of the write interest

### CODE REVIEW

File: libs/nio/src/main/java/org/elasticsearch/nio/NioSelector.java [hunk 1]
```diff
@@ -196,7 +197,8 @@ void cleanupAndCloseChannels() {
         cleanupPendingWrites();
         channelsToClose.addAll(channelsToRegister);
         channelsToRegister.clear();
-        channelsToClose.addAll(selector.keys().stream().map(sk -> (ChannelContext<?>) sk.attachment()).collect(Collectors.toList()));
+        channelsToClose.addAll(selector.keys().stream()
```
Comment (original-brownbear) [inline libs/nio/src/main/java/org/elasticsearch/nio/NioSelector.java]: This started failing when a close was executed before the attachment was set on the selection key. It seems to me that this just got more likely with this change timing wise maybe, but was completely possible without it already.

**Comment by original-brownbear** [comment]
@tbrooks8 ping, not urgent just for when you have a sec :)

**Review by Tim-Brooks — APPROVED** [review]
LGTM

**Comment by original-brownbear** [comment]
thanks @tbrooks8 !

**Pull request merged**